DbService/src/DbService_service.cc

```diff
@@ -61,22 +61,29 @@ namespace mu2e {
     } // end loop over files
 
     int cacheLifetime = 0;
-    _config.cacheLifetime(cacheLifetime);
-    _engine.reader().setCacheLifetime(cacheLifetime);
+    if(_config.cacheLifetime(cacheLifetime)) {
+      _engine.reader().setCacheLifetime(cacheLifetime);
+    }
+
+    int retryTimeout = 0;
+    if(_config.retryTimeout(retryTimeout)) {
+      _engine.reader().setTimeout(retryTimeout);
+    }
+
 
     // service will start calling the database at the first event,
     // so the service can exist without the DB being contacted.  
     // fastStart overrides this and starts reading the DB imediately.
     bool fastStart = false;
-    _config.fastStart(fastStart);
-    if (fastStart) {
-      // this prepares IOV infrastructure for efficient queries
-      if(_verbose>2) std::cout << "DbService::beginJob initializing engine "
-			       <<std::endl;
-      _engine.beginJob();
+    if(_config.fastStart(fastStart)) {
+      if (fastStart) {
```

Comment:
I think that this test is redundant.  The call to _config.fastStart(argument) both returns true/false and writes the return value into the argument.

Reply:
I think the expression in the if tests if the optional parameter was present.
It is a little redundant in that if the parameter was not present, then the 
fastStart variable will not be set in the conf call, and it will have the value I initialized it
to, which I could use predictably.  I wrote this way to be parallel other optimal parameters that are not
bool's.  For example, a present, but empty sequence might be treated differently
than a missing optional parameter.  If is just a little more logically organized
to test if the parameter was input before using it.